docs/manual/scala/guide/services/ServiceClients.md

```diff
@@ -14,6 +14,14 @@ Having bound the client, you can now use it anywhere in your Lagom application.
 
 @[hello-consumer](code/ServiceClients.scala)
 
+## Streaming service client configuration
+
+When using a streaming service client, Lagom will use internally a WebSocket client which has a max frame length parameter. This parameter limits the allowed maximum size for the messages flowing through the WebSocket. This can be configured in `application.conf` on the client-side and the default configuration is:
+
+@[web-socket-client-default](../../../../../service/core/client/src/main/resources/reference.conf)
+
+Note that the same parameter has to be configured on the server-side using [Play server configuration](https://www.playframework.com/documentation/2.6.x/ScalaWebSockets#Configuring-WebSocket-Frame-Length)
```

Comment:
Hmm, does that mean both ends need to be in sync? What's the outcome if they are not? If both ends need to be in sync then we may be better off configuring this setting on the Service Descriptor (if not it's fine as it is).

Also, how would a rolling upgrade be impacted? By rolling upgrade I mean a client side were some client instances use one value for `lagom.client.websocket.frame.maxLength` and the rest use the default. Or similarly, in the server side.

I think it's worth mentioning whether this setting is per endpoint, per target service or per client instance. I understand it's the later so if a Service `A` consumes websockets from `B` and `C` and service `A` configures the `maxLength` to `1024` that would impact communication to both `B` and `C`. This is slightly different than  the settings we offer for Circuit-breakers where each single circuit breaker may override the defaults so I think it's worth mentioning.

Reply:
Good questions.

I'm not an expert on that topic but as I understand it, there is no need to be in sync with the server. This is a protection against some attacks that may be, or not be, implemented in particular client or server implementations. And actually it could be "useful" to have different values on each side as it is not the same type of messages that flow in one direction or the other.

As far as I know, contrary to the netty one, the akka-http websocket client does not have such a parameter (and thus I guess it does not implement such a protection mechanism). Actually I am wondering whether it is really useful to have such a protection on the client side...but maybe, I missed somehting...

Reply:
For the rolling upgrade question, I do not see any particular issue with this value.

Indeed, the parameter is per client instance and that's true that if it consumes multiples websockets, they will all be impacted. And when I think of it, that's not a good behavior. What do you think? Would it better to use the same mechanism than for the circuit-breakers, i.e. to set a websocket identifier in the service descriptor, which could then be used to the configure the frame length on the client side ? Or is it better to keep it as it is and simply document that fact?

Reply:
Regarding my question, I note that on the server side, it would be difficult to have this parameter at the service level as it is handled by Play. So, it may be confusing if we have a websocket service identification mechanism that could only be used to configure the client side but not on the server side.